common/filesystem.js

```diff
@@ -0,0 +1,23 @@
+const fs = require('node:fs');
```

Comment:
Similarly, this module can just be inlined; just define a `function searchAncestors() { ... }` in the main module.

Reply:
Also similarly, this is being used to locate the roots of the packages templates reside in during stage 3 and stage 4. (In fact, I just copied [this function](https://github.com/benblank/ember-first-class-component-templates-migrator/blob/fd5f0d6f423e1cb173fde120147a48466dfa0821/transforms/stage-3-wheres-waldo/index.js#L64) from my WIP stage 3 branch to start with.)

I expect stage 2 will end up needing it, too, though I'll probably end up specializing back into `findPackageRoot` once I have a better idea of whether stages 2 or 4 will need it for anything else.